lib/openssl/ccm.rb

```diff
@@ -21,10 +21,7 @@ class CCM
     #
     # @return [[String]] supported algorithms
     def self.ciphers
-      l = OpenSSL::Cipher.ciphers.keep_if { |c| c.end_with?('-128-CBC') or
-        c.end_with?('-192-CBC') or c.end_with?('-256-CBC') }
-      l.length.times { |i| l[i] = l[i][0..-9] }
-      l
+      @ciphers ||= OpenSSL::Cipher.ciphers.select { |c| c.match(/-(128|192|256)-CBC$/i) }.map { |e| e[0..-9] }.uniq
```

Comment:
hey @franee,

Notice the tailing `/i` in `.match(/-(128|192|256)-CBC$/i)`, that represents case-insensitive regex matching.

Can u provide the URL to the failing build.

The specs passed in my local environment. Looking at the [TravisCI build](https://travis-ci.org/SmallLars/openssl-ccm/builds/474689946), it seems the failure is unrelated to my changes. 🤔 (or perhaps I'm missing it?)

![image](https://user-images.githubusercontent.com/8832106/50633523-8b61fd80-0f86-11e9-8f22-dc222f31ca5c.png)

Reply:
I see you're coming from [adyen-cse-ruby](https://github.com/jooeycheng/adyen-cse-ruby).

I've created https://github.com/jooeycheng/adyen-cse-ruby/issues/11 to track this, so other users who are facing similar issues know what's up.

Reply:
@jooeycheng all calls to `assert(OpenSSL::CCM.ciphers.include?('AES'), 'Missing AES-Cipher')` are failing since CCM.ciphers only has lower-cased strings in it (depends on the openssl version also as what you initially reported).

forked with my tweaks to make tests pass https://github.com/franee/openssl-ccm

Reply:
@franee oh yes, you are right. 🤦‍♂️
I failed to see your point earlier.

Reply:
@jooeycheng Can you add the upcase and squash the commits to one?

Reply:
@SmallLars done 👌